roles/load-ycsb/tasks/main.yml

```diff
@@ -0,0 +1,66 @@
+---
+- name: setting ycsb_workload var
+  set_fact:
+    ycsb_workload: "workloada"
+
+- name: Generate workloads
+  k8s:
+    definition: "{{ lookup('template', 'configmap.yml.j2') | from_yaml }}"
+
+- name: Load Data Into Database
+  k8s:
+    definition:
+      kind: Job
+      apiVersion: batch/v1
+      metadata:
+        name: '{{ meta.name }}-ycsb-data-load-job-{{ ycsb_workload }}'
+        namespace: '{{ operator_namespace }}'
+      spec:
+        template:
+          metadata:
+            name: '{{ meta.name }}-ycsb-data-load'
+            namespace: '{{ operator_namespace }}'
+            labels:
+             name: 'ycsb-load'
+          spec:
+            containers:
+            - name: ycsb
+              image: "quay.io/cloud-bulldozer/ycsb-server:latest"
+              env:
+              - name: es
+                value: "{{elasticsearch.server | default()}}"
+              - name: es_port
+                value: "{{elasticsearch.port | default()}}"
+              - name: user
+                value: "{{test_user | default()}}"
+              - name: workload
+                value: "{{ycsb_workload}}"
+              - name: num_records
+                value: "{{ycsb.recordcount}}"
+              - name: num_operations
+                value: "{{ycsb.operationcount}}"
+              - name: uuid
+                value: "{{uuid}}"
+              imagePullPolicy: Always
+              command: ["/bin/sh"]
+              args: ["-c", "/opt/snafu/ycsb-wrapper/ycsb-wrapper.py -r 1 -l -d {{ ycsb.driver }} -w {{ ycsb_workload }} -x \"{{ ycsb.options_load }}\""]
+              volumeMounts:
+                - name: config-volume
+                  mountPath: "/tmp/ycsb"
+            restartPolicy: Never
+            volumes:
+            - name: config-volume
+              configMap:
+                name: ycsb-test
+        backoffLimit: 4
```

Comment:
Do you want to pull this job out into a template to clean up the task file a bit?

Reply:
I thought about this. It might not be a bad idea.